### PR TITLE
Am/chore/do not run bench on internal

### PR DESCRIPTION
**Pull request by IceTDrinker**

### CODE REVIEW

**Review by agnesLeroy — APPROVED** [review]

File: .github/workflows/aws_tfhe_integer_gpu_tests.yml [hunk 1]
```diff
@@ -59,7 +59,8 @@ jobs:
   setup-instance:
     name: Setup instance (cuda-unsigned-integer-tests)
     needs: should-run
-    if: github.event_name != 'pull_request' ||
+    if: (github.event_name == 'schedule' && github.repository == 'zama-ai/tfhe-rs') ||
+      github.event_name == 'workflow_dispatch' ||
```
Comment (agnesLeroy) [inline .github/workflows/aws_tfhe_integer_gpu_tests.yml:63]: Hmm wait this is the tests, not the bench

File: .github/workflows/aws_tfhe_signed_integer_gpu_tests.yml [hunk 2]
```diff
@@ -61,7 +61,8 @@ jobs:
     name: Setup instance (cuda-signed-integer-tests)
     runs-on: ubuntu-latest
     needs: should-run
-    if: github.event_name != 'pull_request' ||
+    if: (github.event_name == 'schedule' && github.repository == 'zama-ai/tfhe-rs') ||
+      github.event_name == 'workflow_dispatch' ||
```
Comment (agnesLeroy) [inline .github/workflows/aws_tfhe_signed_integer_gpu_tests.yml:65]: Same here, I don't think we want to change this.

File: .github/workflows/gpu_4090_full_benchmark.yml [hunk 3]
```diff
@@ -24,7 +24,9 @@ on:
 jobs:
   cuda-integer-benchmarks:
     name: Cuda integer benchmarks for all operations flavor  (RTX 4090)
-    if: ${{ github.event_name == 'workflow_dispatch' || github.event_name == 'schedule' || contains(github.event.label.name, '4090_bench') }}
+    if: ${{ github.event_name == 'workflow_dispatch' ||
+      github.event_name == 'schedule' && github.repository == 'zama-ai/tfhe-rs' ||
+      contains(github.event.label.name, '4090_bench') }}
```
Comment (agnesLeroy) [inline .github/workflows/gpu_4090_full_benchmark.yml:29]: Actually we should do this on all bench workflows, not just the rtx one

File: .github/workflows/aws_tfhe_signed_integer_gpu_tests.yml (hunk 2 above)
Comment (IceTDrinker) [reply .github/workflows/aws_tfhe_signed_integer_gpu_tests.yml:65]: no we do, this runs the GPU tests on internal, I started the branch with the 4090 bench hence the name branch that is a bit confusing

File: .github/workflows/gpu_4090_full_benchmark.yml (hunk 3 above)
Comment (IceTDrinker) [reply .github/workflows/gpu_4090_full_benchmark.yml:29]: I did not see non RTX benches on internal

File: .github/workflows/aws_tfhe_signed_integer_tests.yml [hunk 4]
```diff
@@ -31,8 +31,9 @@ jobs:
   setup-instance:
     name: Setup instance (signed-integer-tests)
     if: (github.event_name == 'push' && github.repository == 'zama-ai/tfhe-rs') ||
+        (github.event_name == 'schedule' && github.repository == 'zama-ai/tfhe-rs') ||
         (github.event_name == 'pull_request' && contains(github.event.label.name, 'approved')) ||
-        (github.event_name != 'push' && github.event_name != 'pull_request')
+        github.event_name != 'workflow_dispatch'
```
Comment (IceTDrinker) [inline .github/workflows/aws_tfhe_signed_integer_tests.yml:36]: this is a typo

**Review by agnesLeroy — APPROVED** [review]
Looks good to me!

**Pull request merged**